### PR TITLE
Added ✅ : josephus problem

### LINKED ISSUE #2622: Add Josephus Problem in python
To request a new code snippet, please fill out the following:

Project name: Josephus Problem
Project link: https://sampleprograms.io/projects/my-project-name
Language: Python

If this code snippet will also be the first in a new language, please fill out the following:

Official Language Style Guide: 
Official Language Website: 
Official Language Docker Image: https://hub.docker.com/r/repository/image 

> The project link must be a link to a project on the [Sample Programs Website project list][sample-programs-project-list].
> If you would like to add a new project to the Sample Programs Website,
> please first make a pull request to the [Sample Programs Website][sample-programs-website].

[sample-programs-website]: https://github.com/TheRenegadeCoder/sample-programs-website
[sample-programs-project-list]: https://sampleprograms.io/projects/

### PR DESCRIPTION
**Pull request by srini047**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [x] I fixed #2622 
- [x] I named the pull request using `Add `josephus_problem` in `.py`` format
  
  
## I Am Adding a New Code Snippet in a New Language

- [ ] I fixed #your-issue-number-here
- [ ] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format
- [ ] I added a `testinfo.yml` files (see [contributing documentation][contributing-new-language])
  - [ ] I used an officially supported docker image or one that I personally trust
  

## I Am Adding New Tests for a Project

- [ ] I fixed #your-issue-number-here
- [ ] I named the pull request using `Add {PROJECT} tests` format
- [ ] I verified the project exists on the [Sample Programs Project List][sample-programs-project-list]
- [ ] I added tests for every test case in the Testing table of the project description for the given project. (See [contributing documentation][contributing-tests-in-detail])

## I Am Adding New Tests for a Language

- [ ] I fixed #your-issue-number-here
- [ ] I named the pull request using `Add {LANGUAGE} tests` format\
- [ ] I added a `testinfo.yml` files (see [contributing documentation][contributing-new-language])
  - [ ] I used an officially supported docker image or one that I personally trust
- [ ] I verified all tests are passing

  
## I Am Modifying an Existing Code Snippet or Existing Tests

- [ ] I fixed #your-issue-number-here
- [ ] I have read about modifying existing code snippets or tests in the [contributing documentation][contributing-modifications]
- [ ] I have explained why this modification is _necessary_ either in the issue linked above or in the notes below
- [ ] I named the pull request using `Modify {PROJECT} {tests} in {LANGUAGE}` format

## Other Notes

Add notes here...

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sampleprograms.io/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

### CODE REVIEW

**Review by jrg94 — CHANGES_REQUESTED** [review]
Really great start! I would take a peek at the docs: https://sampleprograms.io/projects/josephus-problem/

n and k should come from the command line (i.e., sys.argv).

**Comment by srini047** [comment]
Done with the necessary changes @jrg94

**Review by jrg94 — APPROVED** [review]
Looks awesome! The last print might be incorrect, but otherwise this looks good. Going to let the tests run once more before I officially merge.

**Comment by srini047** [comment]
@jrg94 any updates on the test.

**Comment by jrg94** [comment]
Yes, sorry! I've been on vacation. Here are the test results (seems like a quick fix):

```
=================================== FAILURES ===================================
______ test_josephus_problem_valid[josephus_problem.py-sample input 5, 2] ______

description = 'sample input 5, 2', in_params = '5 2', expected = '3'
josephus_problem = Source(name: josephus_problem, path: /home/runner/work/sample-programs/sample-programs/archive/p/python)

    @project_test(ProjectType.JosephusProblem.key)
    @pytest.mark.parametrize(valid_permutations[0], valid_permutations[1],
                             ids=[p[0] for p in valid_permutations[1]])
    def test_josephus_problem_valid(description, in_params, expected, josephus_problem):
        actual = josephus_problem.run(params=in_params)
>       assert actual.replace('[', '').replace(']', '').strip() == expected
E       AssertionError: assert 'Safe place:  3' == '3'
E         - Safe place:  3
E         + 3

test/projects/test_josephus_problem.py:59: AssertionError
______ test_josephus_problem_valid[josephus_problem.py-sample input 7 3] _______

description = 'sample input 7 3', in_params = '7 3', expected = '4'
josephus_problem = Source(name: josephus_problem, path: /home/runner/work/sample-programs/sample-programs/archive/p/python)

    @project_test(ProjectType.JosephusProblem.key)
    @pytest.mark.parametrize(valid_permutations[0], valid_permutations[1],
                             ids=[p[0] for p in valid_permutations[1]])
    def test_josephus_problem_valid(description, in_params, expected, josephus_problem):
        actual = josephus_problem.run(params=in_params)
>       assert actual.replace('[', '').replace(']', '').strip() == expected
E       AssertionError: assert 'Safe place:  4' == '4'
E         - Safe place:  4
E         + 4

test/projects/test_josephus_problem.py:59: AssertionError
_____ test_josephus_problem_valid[josephus_problem.py-sample input 41, 4] ______

description = 'sample input 41, 4', in_params = '41 4', expected = '11'
josephus_problem = Source(name: josephus_problem, path: /home/runner/work/sample-programs/sample-programs/archive/p/python)

    @project_test(ProjectType.JosephusProblem.key)
    @pytest.mark.parametrize(valid_permutations[0], valid_permutations[1],
                             ids=[p[0] for p in valid_permutations[1]])
    def test_josephus_problem_valid(description, in_params, expected, josephus_problem):
        actual = josephus_problem.run(params=in_params)
>       assert actual.replace('[', '').replace(']', '').strip() == expected
E       AssertionError: assert 'Safe place:  11' == '11'
E         - Safe place:  11
E         + 11

test/projects/test_josephus_problem.py:59: AssertionError
================== 3 failed, 2201 passed in 750.15s (0:12:30) ==================
```

File: archive/p/python/josephus_problem.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+except:
+  print("Usage: please input the total number of people and number of people to skip.")
+  exit(1)
+
+if (n <= k):
+    print("Usage: please input the total number of people and number of people to skip.")
+    exit(1)
+
+print("Safe place: ", josephus(n, k))
```
Comment (jrg94) [inline archive/p/python/josephus_problem.py:23]:
```suggestion
print(josephus(n, k))
```

Comment (srini047) [reply archive/p/python/josephus_problem.py]: @jrg94 Changes addressed.

**Pull request merged**